src/main/java/seedu/exercise/model/ModelManager.java

```diff
@@ -293,16 +294,39 @@ public void addCustomProperty(CustomProperty customProperty) {
         return databaseBook;
     }
 
+    public ReadOnlyResourceBook<Exercise> getExerciseDatabaseData() {
+        return databaseBook;
+    }
+
     //=========== Suggested Exercise Accessors ===============================================================
 
     @Override
     public ObservableList<Exercise> getSuggestedExerciseList() {
-        return suggestedExercises;
+        return suggestions;
+    }
+
+    @Override
+    public void setSuggestions(List<Exercise> suggestions) {
+        this.suggestions.setAll(suggestions);
     }
 
     @Override
     public void updateSuggestedExerciseList(Predicate<Exercise> predicate) {
-        suggestedExercises.setPredicate(predicate);
+        List<Exercise> filteredSuggestions = generateAllSuggestions().filtered(predicate);
```

Comment:
can we use assertion here to ensure `predicate` is `non-null` instead of adding a documentation in `Model` interface.

Reply:
ok